internal/extensionapi/connection_route.go

```diff
@@ -2,9 +2,56 @@
 package extensionapi
 
 import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
 	"net/http"
+	"os"
+	"strings"
+
+	"github.com/jupyter-ai-contrib/jupyter-k8s/internal/aws"
+	"github.com/jupyter-ai-contrib/jupyter-k8s/internal/workspace"
+	corev1 "k8s.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
 )
 
+// noOpPodExec is a no-op implementation of PodExecInterface for connection URL generation
+type noOpPodExec struct{}
+
+func (n *noOpPodExec) ExecInPod(ctx context.Context, pod *corev1.Pod, containerName string, cmd []string, stdin string) (string, error) {
+	return "", fmt.Errorf("pod exec not supported in connection URL generation")
+}
+
+// WorkspaceConnectionRequest represents the request body for creating a workspace connection
+type WorkspaceConnectionRequest struct {
+	APIVersion string                         `json:"apiVersion"`
+	Kind       string                         `json:"kind"`
+	Metadata   map[string]interface{}         `json:"metadata"`
+	Spec       WorkspaceConnectionRequestSpec `json:"spec"`
+}
+
+// WorkspaceConnectionRequestSpec represents the spec of a workspace connection request
+type WorkspaceConnectionRequestSpec struct {
+	WorkspaceName           string `json:"workspaceName"`
+	WorkspaceConnectionType string `json:"workspaceConnectionType"`
+}
+
+// WorkspaceConnectionResponse represents the response for a workspace connection
+type WorkspaceConnectionResponse struct {
+	APIVersion string                            `json:"apiVersion"`
+	Kind       string                            `json:"kind"`
+	Metadata   map[string]interface{}            `json:"metadata"`
+	Spec       WorkspaceConnectionRequestSpec    `json:"spec"`
+	Status     WorkspaceConnectionResponseStatus `json:"status"`
+}
+
+// WorkspaceConnectionResponseStatus represents the status of a workspace connection response
+type WorkspaceConnectionResponseStatus struct {
+	WorkspaceConnectionType string `json:"workspaceConnectionType"`
+	WorkspaceConnectionURL  string `json:"workspaceConnectionUrl"`
+}
```

Comment:
Define this [here](https://github.com/jupyter-infra/jupyter-k8s/tree/main/api/connection/v1alpha1) please.

It'll generate the `deepCopy` method automatically for you.

Then you can just `import connectionv1alpha1` as [here](https://github.com/jupyter-infra/jupyter-k8s/blob/fe7fc5dd43f9602ad16758c091407515046f9332/internal/extensionapi/serverroute_connection_access_review.go#L11)